packages/material-ui-styles/src/ThemeProvider/ThemeProvider.test.js

```diff
@@ -1,94 +1,12 @@
 import * as React from 'react';
 import { expect } from 'chai';
+import ThemeProvider from '@material-ui/private-theming/ThemeProvider';
 import { createClientRender } from 'test/utils';
 import makeStyles from '../makeStyles';
-import useTheme from '../useTheme';
-import ThemeProvider from './ThemeProvider';
 
 describe('ThemeProvider', () => {
   const render = createClientRender();
 
-  it('should provide the theme', () => {
-    const ref = React.createRef();
-    const text = () => ref.current.textContent;
-    function Test() {
-      const theme = useTheme();
-
-      return <span ref={ref}>{theme.foo}</span>;
-    }
-
-    render(
-      <ThemeProvider theme={{ foo: 'foo' }}>
-        <Test />
-      </ThemeProvider>,
-    );
-    expect(text()).to.equal('foo');
-  });
-
-  it('should merge the themes', () => {
-    const ref = React.createRef();
-    const text = () => ref.current.textContent;
-    function Test() {
-      const theme = useTheme();
-
-      return (
-        <span ref={ref}>
-          {theme.foo}
-          {theme.bar}
-        </span>
-      );
-    }
-
-    render(
-      <ThemeProvider theme={{ bar: 'bar' }}>
-        <ThemeProvider theme={{ foo: 'foo' }}>
-          <Test />
-        </ThemeProvider>
-      </ThemeProvider>,
-    );
-    expect(text()).to.equal('foobar');
-  });
-
-  it('should memoize the merged output', () => {
-    const themes = [];
-
-    const ref = React.createRef();
-    const text = () => ref.current.textContent;
-    function Test() {
-      const theme = useTheme();
-      React.useEffect(() => {
-        themes.push(theme);
-      });
-
-      return (
-        <span ref={ref}>
-          {theme.foo}
-          {theme.bar}
-        </span>
-      );
-    }
-    const MemoTest = React.memo(Test);
-
-    const outerTheme = { bar: 'bar' };
-    const innerTheme = { foo: 'foo' };
-
-    function Container() {
-      return (
-        <ThemeProvider theme={outerTheme}>
-          <ThemeProvider theme={innerTheme}>
-            <MemoTest />
-          </ThemeProvider>
-        </ThemeProvider>
-      );
-    }
-
-    const { setProps } = render(<Container />);
-    expect(text()).to.equal('foobar');
-    setProps({});
-    expect(text()).to.equal('foobar');
-    expect(themes).to.have.length(1);
-  });
-
   it('does not allow setting mui.nested manually', () => {
```

Comment:
All tests except this one are moved to the `@material-ui/private-theming` package. This test had to stay here as it uses the `makeStyles()` utility which is not available there.